servicetalk-data-protobuf-jersey/src/main/java/io/servicetalk/data/protobuf/jersey/ProtobufSerializerMessageBodyReaderWriter.java

```diff
@@ -0,0 +1,243 @@
+ * limitations under the License.
+ */
+package io.servicetalk.data.protobuf.jersey;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.data.protobuf.ProtobufSerializerFactory;
+import io.servicetalk.http.router.jersey.internal.SourceWrappers.PublisherSource;
+import io.servicetalk.http.router.jersey.internal.SourceWrappers.SingleSource;
+import io.servicetalk.serializer.api.Deserializer;
+import io.servicetalk.serializer.api.SerializationException;
+import io.servicetalk.serializer.api.Serializer;
+import io.servicetalk.serializer.api.StreamingSerializer;
+import io.servicetalk.transport.api.ConnectionContext;
+import io.servicetalk.transport.api.ExecutionContext;
+
+import com.google.protobuf.MessageLite;
+import org.glassfish.jersey.internal.util.collection.Ref;
+import org.glassfish.jersey.server.ExtendedUriInfo;
+import org.glassfish.jersey.server.model.ResourceMethod;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.NoSuchElementException;
+import javax.annotation.Priority;
+import javax.inject.Provider;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.ContextResolver;
+import javax.ws.rs.ext.MessageBodyReader;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.Providers;
+
+import static io.servicetalk.concurrent.api.Publisher.fromInputStream;
+import static io.servicetalk.concurrent.internal.FutureUtils.awaitResult;
+import static io.servicetalk.data.protobuf.jersey.ProtobufMediaTypes.APPLICATION_PROTOBUF;
+import static io.servicetalk.data.protobuf.jersey.ProtobufMediaTypes.APPLICATION_PROTOBUF_TYPE;
+import static io.servicetalk.data.protobuf.jersey.ProtobufMediaTypes.APPLICATION_PROTOBUF_VAR_INT;
+import static io.servicetalk.data.protobuf.jersey.ProtobufMediaTypes.APPLICATION_PROTOBUF_VAR_INT_TYPE;
+import static io.servicetalk.data.protobuf.jersey.ProtobufSerializerMessageBodyReaderWriter.PRODUCES_CONSUMES_STR;
+import static io.servicetalk.http.router.jersey.internal.BufferPublisherInputStream.handleEntityStream;
+import static io.servicetalk.http.router.jersey.internal.RequestProperties.setResponseBufferPublisher;
+import static javax.ws.rs.Priorities.ENTITY_CODER;
+
+// Less priority than the *MessageBodyReaderWriters provided by the Jersey Router itself to avoid attempting
+// Protobuf (de)serialization of core types like Buffer.
+@Priority(ENTITY_CODER + 100)
```

Comment:
How do you think, should we make this priority slightly different than jackson? Not sure how that works when both features are registered.

Reply:
I think starting with the same priority is fine, we can adjust later if necessary.